Dockerfile.dev

```diff
@@ -52,6 +52,7 @@ USER vscode
 ENV VIRTUAL_ENV="/home/vscode/.local/ha-venv"
 RUN uv venv $VIRTUAL_ENV
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+ENV container=oci
```

Comment:
- Same as above
- Also, why is this needed for the dev container?

Reply:
If a developer runs the dev container, they should expect consistency with the non-dev one, so I set the variable in both.

Reply:
Seems impossible to hit. Please provide a path on how one would hit this?

Reply:
@javydekoning I understood from the above comment. I don't use Docker desktop myself either. That said, what would be the path to hit that?

Reply:
A path would be to use contained (which sadly doesn't hit the other criteria per https://github.com/home-assistant/core/pull/132404#issuecomment-2528459790).

Reply:
Done with a634dbf863

Reply:
The thing I'm trying to point out here, is that (as far as I'm aware) you can't run devcontainers via these paths that use these runtimes this PR is about. Therefore it won't be hit by any of this and thus (IMHO) adding a dead env variable.

Reply:
So a developer wouldn't run a `Dockerfile.dev` based container that executes `is_docker_env()`?

If that's true, I agree that removing the change to `Dockerfile.dev` is the right course of action.

Reply:
This Dockerfile is used for the devcontainer specifications used by, for example, vscode & PyCharm. You've added it here, I'm wondering why and how they would hit this path.